chainerx_cc/chainerx/routines/manipulation.cc

```diff
@@ -492,61 +492,14 @@ Array Concatenate(const std::vector<Array>& arrays, nonstd::optional<int8_t> axi
     return ConcatenateImpl(raveled_arrays, 0);
 }
 
-namespace {
-std::vector<Array> StackGrad(const Array& gout, int8_t axis) {
-    Shape shape{gout.shape()};
-    Strides strides{gout.strides()};
-    size_t dim = shape[axis];
-    int64_t step = strides[axis];
-    int64_t offset = gout.offset();
-    bool is_empty = gout.GetTotalSize() == 0;
-    shape.erase(shape.begin() + axis);
-    strides.erase(strides.begin() + axis);
-
-    std::vector<Array> gxs;
-    std::vector<ConstArrayRef> gxs_refs{};
-    gxs.reserve(dim);
-    gxs_refs.reserve(dim);
-    {
-        NoBackpropModeScope scope{};
-        Dtype dtype = gout.dtype();
-        Device& device = gout.device();
-        for (size_t i = 0; i < dim; ++i) {
-            gxs.emplace_back(internal::MakeArray(shape, strides, dtype, device, gout.data(), offset));
-            gxs_refs.emplace_back(gxs.back());
-            if (!is_empty) {
-                offset += step;
-            }
-        }
-    }
-
-    {
-        BackwardBuilder bb{"stack-grad", gout, gxs_refs};
-        if (BackwardBuilder::Target bt = bb.CreateTarget()) {
-            bt.Define([axis](BackwardContext& bctx) {
-                std::vector<Array> ggxs;
-                ggxs.reserve(bctx.output_count());
-                for (size_t i = 0; i < bctx.output_count(); ++i) {
-                    // TODO(imanishi): Check if bctx.output_grad(i) is not nullopt.
-                    ggxs.emplace_back(*bctx.output_grad(i));
-                }
-                bctx.input_grad() = Stack(ggxs, axis);
-            });
-        }
-        bb.Finalize();
-    }
-    return gxs;
-}
-}  // namespace
-
 Array Stack(const std::vector<Array>& arrays, int8_t axis) {
```

Comment:
How about just using `Concatenate`?

Reply:
How about addressing that in a separate PR? I don't necessarily think that it's within the scope of this PR. Although I did refactor the gradient computation in a similar way, but that was mostly to simply the mixed dtype support there.

Reply:
Fixed, PTAL.